### PR TITLE
feat: SelectComposeMailboxView

### PR DESCRIPTION
**Pull request by Lnamw**

### CODE REVIEW

**Comment by Lnamw** [comment]
on hold, waiting for changes from UI designer

File: Mail/Views/New Message/Select Mailbox/CurrentComposeMailboxView.swift [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+struct CurrentComposeMailboxView: View {
+    @LazyInjectService private var accountManager: AccountManager
+    @LazyInjectService private var matomo: MatomoUtils
+    @LazyInjectService private var platformDetector: PlatformDetectable
+
+    @AppStorage(UserDefaults.shared.key(.accentColor)) private var accentColor = DefaultPreferences.accentColor
+
+    @Environment(\.dismiss) private var dismiss
+    @Environment(\.dismissModal) var dismissModal
```
Comment (valentinperignon) [inline Mail/Views/New Message/Select Mailbox/CurrentComposeMailboxView.swift:35]:
```suggestion
    @Environment(\.dismissModal) private var dismissModal
```

File: Mail/Views/New Message/Select Mailbox/CurrentComposeMailboxView.swift [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                    .buttonStyle(.ikPlain)
+                    .ikButtonFullWidth(true)
+                    .controlSize(.large)
+
+                NavigationLink(destination: SelectComposeMailboxView(
+                    composeMessageIntent: $composeMessageIntent,
+                    viewModel: viewModel
+                )) {
+                    Text("Envoyer avec une autre adresse")
+                        .textStyle(.bodyMediumAccent)
```
Comment (valentinperignon) [inline Mail/Views/New Message/Select Mailbox/CurrentComposeMailboxView.swift:73]: You can apply
```swift
.ikButtonFullWidth(true)
.controlSize(.large)
```
on both buttons

File: Mail/Views/New Message/Select Mailbox/CurrentComposeMailboxView.swift [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                }
```
Comment (valentinperignon) [inline Mail/Views/New Message/Select Mailbox/CurrentComposeMailboxView.swift:74]: Maybe we can apply a bottom padding ?
`.padding(.bottom, UIPadding.onBoardingBottomButtons)`

File: Mail/Views/New Message/Select Mailbox/SelectComposeMailboxView.swift [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+               let account = accountManager.account(for: selectedMailbox.userId), viewModel.selectionMade {
+                SelectedMailboxView(account: account, selectedMailbox: selectedMailbox)
+            }
+
+            Button(MailResourcesStrings.Localizable.buttonContinue, action: viewModel.validateMailboxChoice)
+                .buttonStyle(.ikPlain)
+                .controlSize(.large)
+                .ikButtonFullWidth(true)
+        }
+        .padding(.horizontal, value: .medium)
```
Comment (valentinperignon) [inline Mail/Views/New Message/Select Mailbox/SelectComposeMailboxView.swift:68]: We can add a bottom padding here too

File: Mail/Views/New Message/Select Mailbox/CurrentComposeMailboxView.swift [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    @StateObject private var viewModel: SelectComposeMailboxViewModel
+
+    @Binding var composeMessageIntent: ComposeMessageIntent
+
+    init(composeMessageIntent: Binding<ComposeMessageIntent>) {
+        _composeMessageIntent = composeMessageIntent
+        _viewModel = StateObject(wrappedValue: SelectComposeMailboxViewModel(composeMessageIntent: composeMessageIntent))
+    }
+
+    var body: some View {
```
Comment (valentinperignon) [inline Mail/Views/New Message/Select Mailbox/CurrentComposeMailboxView.swift:46]: You can add a regular horizontal padding to the whole view, and then a small padding to the buttons

File: Mail/Views/New Message/Select Mailbox/CurrentComposeMailboxView.swift [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+            Text(MailResourcesStrings.Localizable.composeMailboxCurrentTitle)
+                .textStyle(.header2)
+                .multilineTextAlignment(.center)
+                .padding(.bottom, value: .medium)
+                .padding(.horizontal, value: .regular)
+
+            if let selectedMailbox = viewModel.selectedMailbox,
+               let account = accountManager.account(for: selectedMailbox.userId) {
+                SelectedMailboxView(account: account, selectedMailbox: selectedMailbox)
```
Comment (valentinperignon) [inline Mail/Views/New Message/Select Mailbox/CurrentComposeMailboxView.swift:59]: The SelectedMailboxView should have a chevron (you can check Figma 👍 )

File: Mail/Views/New Message/Select Mailbox/SelectComposeMailboxView.swift [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    @LazyInjectService private var platformDetector: PlatformDetectable
+
+    @AppStorage(UserDefaults.shared.key(.accentColor)) private var accentColor = DefaultPreferences.accentColor
+
+    @Binding var composeMessageIntent: ComposeMessageIntent
+
+    let viewModel: SelectComposeMailboxViewModel
+
+    var body: some View {
+        VStack(spacing: 0) {
```
Comment (valentinperignon) [inline Mail/Views/New Message/Select Mailbox/SelectComposeMailboxView.swift:39]: You can do the same here for the horizontal padding

File: Mail/Views/New Message/Select Mailbox/SelectedMailboxView.swift [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+import SwiftUI
+
+struct SelectedMailboxView: View {
+    @EnvironmentObject private var mailboxManager: MailboxManager
+
+    let account: Account
+    let selectedMailbox: Mailbox
+
+    var body: some View {
+        HStack {
```
Comment (valentinperignon) [inline Mail/Views/New Message/Select Mailbox/SelectedMailboxView.swift:31]: You can specify the spacing here

**Review by valentinperignon — COMMENTED** [review]
Juste a few comments 👍 
Please, pull your branch as I rebased to be up to date with master

**Comment by valentinperignon** [comment]
When we click on the button "Send with a different address", the button "continue" should be disabled

File: Mail/Views/New Message/ComposeMessageIntentView.swift [hunk 9]
```diff
@@ -25,24 +25,31 @@ import SwiftUI
...
+        _composeMessageIntent = State(wrappedValue: composeMessageIntent)
+        self.attachments = attachments
+    }
 
     var body: some View {
-        NBNavigationStack {
+        if composeMessageIntent.shouldSelectMailbox {
+            CurrentComposeMailboxView(composeMessageIntent: $composeMessageIntent)
+        } else {
             if let resolvedIntent = resolvedIntent.wrappedValue {
```
Comment (valentinperignon) [inline Mail/Views/New Message/ComposeMessageIntentView.swift:53]: The if-else should be wrapped in a NavigationView, not only the compose view

**Review by valentinperignon — APPROVED** [review]

**Pull request merged**